source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -463,27 +463,36 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const Envoy::Ssl::ContextConfig& c
 
   parsed_alpn_protocols_ = parseAlpnProtocols(config.alpnProtocols());
 
-  // To enumerate the required builtin ciphers, curves, algorithms, and
-  // versions, uncomment '#define LOG_BUILTIN_STAT_NAMES' below, and run
-  //  bazel test //test/extensions/transport_sockets/tls/... --test_output=streamed
-  //      | grep " Builtin ssl." | sort | uniq
-  // #define LOG_BUILTIN_STAT_NAMES
-  //
-  // TODO(#8035): improve tooling to find any other built-ins needed to avoid
-  // contention.
+  // Use the SSL library to iterate over the configured ciphers.
+  for (TlsContext& tls_context : tls_contexts_) {
+    bssl::UniquePtr<SSL> ssl = bssl::UniquePtr<SSL>(SSL_new(tls_context.ssl_ctx_.get()));
+    STACK_OF(SSL_CIPHER)* ciphers = SSL_get_ciphers(ssl.get());
+    for (uint32_t i = 0, n = sk_SSL_CIPHER_num(ciphers); i < n; ++i) {
+      const SSL_CIPHER* cipher = sk_SSL_CIPHER_value(ciphers, i);
+      stat_name_set_->rememberBuiltin(SSL_CIPHER_get_name(cipher));
+      ENVOY_LOG_MISC(error, SSL_CIPHER_get_name(cipher));
+    }
+  }
 
-  // Ciphers
-  stat_name_set_->rememberBuiltin("AEAD-AES128-GCM-SHA256");
-  stat_name_set_->rememberBuiltin("ECDHE-ECDSA-AES128-GCM-SHA256");
-  stat_name_set_->rememberBuiltin("ECDHE-RSA-AES128-GCM-SHA256");
-  stat_name_set_->rememberBuiltin("ECDHE-RSA-AES128-SHA");
-  stat_name_set_->rememberBuiltin("ECDHE-RSA-CHACHA20-POLY1305");
-  stat_name_set_->rememberBuiltin("TLS_AES_128_GCM_SHA256");
+  // TLS_AES_128_GCM_SHA256 is referenced from
+  // IpVersionsClientVersions/SslCertficateIntegrationTest.ServerRsa/IPv4_TLSv1_3,
+  // possibly due to the call to
+  // ClientSslTransportOptions().setSigningAlgorithmsForTest
+  // in test/extensions/transport_sockets/tls/integration/ssl_integration_test.cc, function
+  // rsaOnlyClientOptions.
+  //
+  // We must also add two more that are hardcoded into the SSL 1.3 spec:
+  // https://tools.ietf.org/html/rfc8446
```

Comment:
Majority of this comment is irrelevant. I'd leave only the note about TLS 1.3 (not SSL!).

Reply:
Cleaned up, thanks.

Reply:
Done.